clippy_lints/src/implied_bounds_in_impls.rs

```diff
@@ -162,10 +249,14 @@ fn check(cx: &LateContext<'_>, decl: &FnDecl<'_>) {
             if let GenericBound::Trait(poly_trait, TraitBoundModifier::None) = bound
                 && let [.., path] = poly_trait.trait_ref.path.segments
                 && let implied_args = path.args.map_or([].as_slice(), |a| a.args)
+                && let implied_bindings = path.args.map_or([].as_slice(), |a| a.bindings)
                 && let Some(def_id) = poly_trait.trait_ref.path.res.opt_def_id()
-                && let Some(implied_by_span) = implied_bounds
+                && let Some((implied_by_span, implied_by_args, implied_by_bindings)) = implied_bounds
                     .iter()
-                    .find_map(|&(span, implied_by_args, preds, implied_by_def_id)| {
+                    .find_map(|&(span, implied_by_path, preds, implied_by_def_id)| {
+                        let implied_by_args = implied_by_path.args.map_or([].as_slice(), |a| a.args);
+                        let implied_by_bindings = implied_by_path.args.map_or([].as_slice(), |a| a.bindings);
```

Comment:
As these `.as_slice()` do not convey useful info but clutter the code, I think we can replace them with `&[]` (IIRC, `&[]` is equivalent to `[].as_slice()`)

Reply:
This sadly [doesn't work](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=024fdf2625542008e7ffaae2fa912466).

 `&[]` is more like of type `&[_; 0]`, which can usually coerce to `&[_]`, but not here (probably because it's in a generic context), and `[].as_slice()` does the coercion explicitly to avoid the type check error. We could also do `&[] as &[_]`, but imo it looks weird and it's not that much better than `[].as_slice()`